### PR TITLE
chore(extension-api): enhance health check

### LINKED ISSUE #6295: docs(extension-api): enhance health check
### Is your enhancement related to a problem? Please describe

Not enough information on how to use health check when creating containers 

### Describe the solution you'd like

adding some documentation

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding some documentation for the Health checks

**sources**
- https://developers.redhat.com/blog/2019/04/18/monitoring-container-vitality-and-availability-with-podman
- [containers/podman#libpod/define/container_inspect.go](https://github.com/containers/podman/blob/56e0f063bd280318a861d1c7f816160c31fdf7bb/libpod/define/container_inspect.go#L290)

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop/issues/6295

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -2321,35 +2352,65 @@ declare module '@podman-desktop/api' {
...
+     * Test?: ["CMD", "curl", "http://localhost"];
+     *
+     * @example
+     * // Podman will execute the command inside the target container and wait for either a "0" or "failure  exit" code.
+     * Test?: ["CMD-SHELL", "curl http://localhost || exit 1"];
      */
     Test?: string[];
 
     /**
-     * The time to wait between checks in nanoseconds. It should be 0 or at least 1000000 (1 ms). 0 means inherit.
```
Comment (feloy) [inline packages/extension-api/src/extension-api.d.ts:2336]: Why removing the min value, and, more useful, the 0 value meaning?

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -2321,35 +2352,65 @@ declare module '@podman-desktop/api' {
...
+     * The time to wait between checks in nanoseconds.
+     *
+     * @example
+     * // Set interval to 1 second
+     * Interval?: 1000000000;
      */
     Interval?: number;
 
     /**
-     * The time to wait before considering the check to have hung. It should be 0 or at least 1000000 (1 ms). 0 means inherit.
```
Comment (feloy) [inline packages/extension-api/src/extension-api.d.ts:2341]: same question here, and below

File: packages/extension-api/src/extension-api.d.ts [hunk 3]
```diff
@@ -2321,35 +2352,65 @@ declare module '@podman-desktop/api' {
...
+     *
+     * @example
+     * // Podman will execute the command inside the target container and wait for either a "0" or "failure  exit" code.
+     * Test?: ["CMD-SHELL", "curl http://localhost || exit 1"];
      */
     Test?: string[];
 
     /**
-     * The time to wait between checks in nanoseconds. It should be 0 or at least 1000000 (1 ms). 0 means inherit.
+     * The time to wait between checks in nanoseconds.
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
     * The time to wait between checks in nanoseconds. It should be 0 or at least 1000000 (1 ms). 0 means inherit.
```

File: packages/extension-api/src/extension-api.d.ts [hunk 4]
```diff
@@ -2321,35 +2352,65 @@ declare module '@podman-desktop/api' {
...
+     *
+     * @example
+     * // Set interval to 1 second
+     * Interval?: 1000000000;
      */
     Interval?: number;
 
     /**
-     * The time to wait before considering the check to have hung. It should be 0 or at least 1000000 (1 ms). 0 means inherit.
+     * The time to wait before considering the check to have hung.
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
     * The time to wait before considering the check to have hung. It should be 0 or at least 1000000 (1 ms). 0 means inherit.
```

File: packages/extension-api/src/extension-api.d.ts [hunk 5]
```diff
@@ -2321,35 +2352,65 @@ declare module '@podman-desktop/api' {
...
+     * @example
+     * // Set timeout to 5 seconds
+     * Timeout?: 5000000000;
      */
     Timeout?: number;
 
     /**
-     * Start period for the container to initialize before starting health-retries countdown in nanoseconds. It should
-     * be 0 or at least 1000000 (1 ms). 0 means inherit.
+     * Start period for the container to initialize before starting health-retries countdown in nanoseconds.
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
     * Start period for the container to initialize before starting health-retries countdown in nanoseconds. It should
     * be 0 or at least 1000000 (1 ms). 0 means inherit.
```

File: packages/extension-api/src/extension-api.d.ts [hunk 6]
```diff
@@ -2321,35 +2352,65 @@ declare module '@podman-desktop/api' {
...
+     *
+     * @example
+     * // Set start period to 2 seconds
+     * StartPeriod?: 2000000000;
      */
     StartPeriod?: number;
 
     /**
-     * The number of consecutive failures needed to consider a container as unhealthy. 0 means inherit.
+     * The number of consecutive failures needed to consider a container as unhealthy.
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
     * The number of consecutive failures needed to consider a container as unhealthy. 0 means inherit.
```

File: packages/extension-api/src/extension-api.d.ts (hunk 1 above)
Comment (axel7083) [reply packages/extension-api/src/extension-api.d.ts:2336]: Probably a bad copy past, I restored the proper version

**Review by feloy — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Pull request merged**